lib/workers/repository/cache.ts

```diff
@@ -52,8 +47,8 @@ async function generateBranchCache(
 ): Promise<BranchCache | null> {
   const { baseBranch, branchName } = branch;
   try {
-    const sha = getBranchCommit(branchName) ?? null;
-    const baseBranchSha = getBranchCommit(baseBranch);
+    const sha = (await scm.getBranchCommit(branchName)) ?? null;
```

Comment:
```suggestion
    const sha = await scm.getBranchCommit(branchName);
```
the else should never happen 🤔